core/services/relay/evm/read/event.go

```diff
@@ -270,11 +293,28 @@ func (b *EventBinding) GetLatestValue(ctx context.Context, address common.Addres
 		}
 	}
 
-	return b.decodeLog(ctx, log, into)
+	if err := b.decodeLog(ctx, log, into); err != nil {
+		encoded := hex.EncodeToString(log.Data)
+		result = &encoded
+
+		return err
+	}
+
+	return nil
 }
 
-func (b *EventBinding) QueryKey(ctx context.Context, address common.Address, filter query.KeyFilter, limitAndSort query.LimitAndSort, sequenceDataType any) ([]commontypes.Sequence, error) {
-	if err := b.validateBound(address); err != nil {
+func (b *EventBinding) QueryKey(ctx context.Context, address common.Address, filter query.KeyFilter, limitAndSort query.LimitAndSort, sequenceDataType any) (sequences []commontypes.Sequence, err error) {
+	defer func() {
+		if err != nil {
+			err = newErrorFromCall(err, Call{
+				ContractAddress: address,
+				ContractName:    b.contractName,
+				MethodName:      b.eventName,
```

Comment:
Call should have its MethodName field renamed to ReadName. That way this error creation method won't be ambiguous, alsoBatchCall will most likely be extended to work with events too at some point so changing methodName to ReadName should be alright.